x/upgrade/types/plan.go

```diff
@@ -45,3 +47,25 @@ func (p Plan) ShouldExecute(ctx sdk.Context) bool {
 func (p Plan) DueAt() string {
 	return fmt.Sprintf("height: %d", p.Height)
 }
+
+// UpgradeDetails is a struct that represents the details of an upgrade
+// This is held in the Info object of an upgrade Plan
+type UpgradeDetails struct {
+	UpgradeType string `json:"upgradeType"`
+}
+
+// UpgradeDetails parses and returns a details struct from the Info field of a Plan
+// The upgrade.pb.go is generated from proto, so this is separated here
+func (p Plan) UpgradeDetails() UpgradeDetails {
+	var details UpgradeDetails
+	if err := json.Unmarshal([]byte(p.Info), &details); err != nil {
+		// invalid json, assume no upgrade details
```

Comment:
Can we add an error log here for failed to unmarshal upgrade info so that we know something bad is happening??